light/rpc/client.go

```diff
@@ -324,9 +324,9 @@ func (c *Client) Block(ctx context.Context, height *int64) (*ctypes.ResultBlock,
 	if err := res.Block.ValidateBasic(); err != nil {
 		return nil, err
 	}
-	if bmH, bH := res.BlockID.Hash, res.Block.Hash(); !bytes.Equal(bmH, bH) {
-		return nil, fmt.Errorf("blockID %X does not match with block %X",
-			bmH, bH)
+	rbID := types.BlockID{Hash: res.Block.Hash(), PartSetHeader: res.Block.MakePartSet(types.BlockPartSizeBytes).Header()}
+	if !res.BlockID.Equals(rbID) {
+		return nil, fmt.Errorf("blockID %v does not match with block %v", res.BlockID.String(), rbID.String())
```

Comment:
I don't quite understand how this fix affected the part of here and why it needs to be fixed. Could you tell me how it has affected this?

Reply:
This is the process by which the light client gets the block via rpc. Again, PartSetHeader validation has been added to handle blocks safely as well.

Reply:
The following tests are the same.
```go
if !bytes.Equal(trustedBlockID.Hash, targetBlockID.Hash) {
    panic("invalid header")
}
if !trustedBlockID.PartSetHeader.Equals(targetBlockID.PartSetHeader) {
    panic("invalid entropy")
}
```
```go
if !trustedBlockID.Equals(targetBlockID) {
    panic("invalid block id")
}
```

Reply:
I understand, thanks.